### PR TITLE
feat(configurable-thresholds): Introduce an auto/number type to allow new apdex in discover

### PR DESCRIPTION
**Pull request by shruthilayaj**
Allow the user to type auto or a number into the new
apdex, user misery and count miserable fields so they
may use project level thresholds in the Discover Saved
Query Column Editor Modal.

There is some general weirdness around the columnEditModal
that I've documented in this [ticket](https://getsentry.atlassian.net/jira/software/c/projects/VIS/boards/93/roadmap?selectedIssue=VIS-1070) that will be addressed in
a follow up PR since it exists independent of any changes
in here.

### CODE REVIEW

**Review by wmak — COMMENTED** [review]
This LGTM, I like that we're still validating the text field correctly too so users can't just write random text. Just two concerns:
1. Could we go with `Automatic`? 
2. Looks like `apdex(auto)` is still being picked up by the autocomplete:
image

File: static/app/views/eventsV2/table/queryField.tsx [hunk 1]
```diff
@@ -176,12 +180,20 @@ class QueryField extends React.Component<Props> {
     this.triggerChange(newColumn);
   };
 
-  handleScalarParameterChange = (value: string) => {
-    const newColumn = cloneDeep(this.props.fieldValue);
-    if (newColumn.kind === 'function') {
-      newColumn.function[1] = value;
-    }
-    this.triggerChange(newColumn);
+  handleScalarParameterChange = (dataType: DataType) => {
```
Comment (Zylphrex) [inline static/app/views/eventsV2/table/queryField.tsx:183]: You can actually just chain the arrow functions like

```javascript
(dataType: DataType) => (value: string) => {
   // insert function body here
}
```

File: static/app/views/eventsV2/table/queryField.tsx [hunk 2]
```diff
@@ -605,9 +630,7 @@ class BufferedInput extends React.Component<InputProps, InputState> {
   };
 
   handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    if (this.isValid) {
```
Comment (Zylphrex) [inline static/app/views/eventsV2/table/queryField.tsx:608]: why was this removed?

**Comment by shruthilayaj** [comment]
> This LGTM, I like that we're still validating the text field correctly too so users can't just write random text. Just two concerns:
> 
> 1. Could we go with `Automatic`?
> 2. Looks like `apdex(auto)` is still being picked up by the autocomplete:
> 
> image

Addressed # 2 in d655496a6ae07bd2b5cba86a335f00f6962d82e0
for the first comment, do you mean like `apdex(automatic)` or `user_misery(automatic)` etc?

**Comment by shruthilayaj** [comment]
I personally prefer `auto` since it's shorter if you meant it in the context of what goes in between the parentheses `apdex(...)`

File: static/app/views/eventsV2/table/queryField.tsx [hunk 3]
```diff
@@ -144,7 +144,11 @@ class QueryField extends React.Component<Props> {
             fieldValue.function[i + 2] = undefined;
           }
         } else if (param.kind === 'value') {
-          fieldValue.function[i + 1] = param.defaultValue || '';
+          if (param.dataType === 'autoNumber' && param.defaultValue === 'auto') {
+            fieldValue.function[i + 1] = '';
```
Comment (Zylphrex) [inline static/app/views/eventsV2/table/queryField.tsx:148]: This is turning `'auto'` into `''` right because the backend is expecting `apdex()` with no arguments right? Can we have a comment explaining this?

File: static/app/views/eventsV2/table/queryField.tsx (hunk 2 above)
Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:608]: since the regex is `auto|[0-9]*(\.[0-9]*)?` typing letters like `a` or `au` are considered invalid so the field won't have its state set. The other option was to make the regex something like `auto|aut|au|a|[0-9]*(\.[0-9]*)?` so that the intermediate states are valid, but that's kinda gross

File: static/app/views/eventsV2/table/queryField.tsx (hunk 1 above)
Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:183]: ooh nice lol I was thinking in python decorator mode

File: static/app/views/eventsV2/table/queryField.tsx (hunk 2 above)
Comment (Zylphrex) [reply static/app/views/eventsV2/table/queryField.tsx:608]: This breaks the existing validation on `number` and `integer` inputs 🤔 an actual example would be that it allows you to enter `asdf` as the second argument to `percentile` now, but because of the way `handleBlur` is implemented it resets the value correctly.

I'm a personally little torn on this. If someone else has some thoughts, feel free to weigh in.

Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:608]: yeah, basically `handleBlur` handles the validation now

Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:608]: I'm also not 100% good on this, open to better suggestions

Comment (shruthilayaj) [reply static/app/views/eventsV2/table/queryField.tsx:608]: but on the other hand field validation does happen as soon as you click out of the field which doesn't seem like the worst thing. Especially because all the inputs into these fields are relatively short like 3-4 characters so you're not losing much by not getting real time validation as you type. anything longer than 3-4 characters (like columns) is in a dropdown

**Comment by shruthilayaj** [comment]
Superseded by https://github.com/getsentry/sentry/pull/26573
